readthedocs/projects/views/public.py

```diff
@@ -291,46 +294,22 @@ def project_download_media(request, project_slug, type_, version_slug):
         uip=get_client_ip(request),
     )
 
-    if settings.DEFAULT_PRIVACY_LEVEL == 'public' or settings.DEBUG:
-
-        storage = get_storage_class(settings.RTD_BUILD_MEDIA_STORAGE)()
-        storage_path = version.project.get_storage_path(
-            type_=type_, version_slug=version_slug,
-            version_type=version.type,
-        )
-        if storage.exists(storage_path):
-            return HttpResponseRedirect(storage.url(storage_path))
-
-        media_path = os.path.join(
-            settings.MEDIA_URL,
-            type_,
-            project_slug,
-            version_slug,
-            '%s.%s' % (project_slug, type_.replace('htmlzip', 'zip')),
-        )
-        return HttpResponseRedirect(media_path)
-
-    # Get relative media path
-    path = (
-        version.project.get_production_media_path(
-            type_=type_,
-            version_slug=version_slug,
-        ).replace(settings.PRODUCTION_ROOT, '/prod_artifacts')
+    storage = get_storage_class(settings.RTD_BUILD_MEDIA_STORAGE)()
+    storage_path = version.project.get_storage_path(
+        type_=type_, version_slug=version_slug,
+        version_type=version.type,
     )
-    content_type, encoding = mimetypes.guess_type(path)
-    content_type = content_type or 'application/octet-stream'
-    response = HttpResponse(content_type=content_type)
-    if encoding:
-        response['Content-Encoding'] = encoding
-    response['X-Accel-Redirect'] = path
-    # Include version in filename; this fixes a long-standing bug
-    filename = '{}-{}.{}'.format(
+    if storage.exists(storage_path):
+        return HttpResponseRedirect(storage.url(storage_path))
+
+    media_path = os.path.join(
+        settings.MEDIA_URL,
+        type_,
         project_slug,
         version_slug,
-        path.split('.')[-1],
+        '%s.%s' % (project_slug, type_.replace('htmlzip', 'zip')),
     )
-    response['Content-Disposition'] = 'filename=%s' % filename
```

Comment:
I think this is actually the logic we want going forward. It lets us specify the actual filename of the download, instead of just linking out to a specific page. Linking back to the media server feels like going backwards, no?

Reply:
I think we can't use a `Content-Disposition` header in a 302 redirect.

If we want to keep this header, I think we will need to proxy the request. On community, it seems we lost that header already when redirecting to Blob storage:

```
$ curl -IL https://readthedocs.org/projects/sphinx-notfound-page/downloads/epub/latest/
HTTP/2 302 
date: Mon, 25 Nov 2019 10:47:35 GMT
content-type: text/html; charset=utf-8
set-cookie: __cfduid=d8ba30f4746be12229d35f371d725f28d1574678854; expires=Wed, 25-Dec-19 10:47:34 GMT; path=/; domain=.readthedocs.org; HttpOnly
content-language: en
location: https://buildmedia.readthedocs.org/media/epub/sphinx-notfound-page/latest/sphinx-notfound-page.epub
vary: Accept-Language, Cookie
x-frame-options: DENY
x-content-type-options: nosniff
x-xss-protection: 1; mode=block
strict-transport-security: max-age=31536000;
x-deity: web01
cf-cache-status: DYNAMIC
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
server: cloudflare
cf-ray: 53b30b9a8aa4ee79-CDG

HTTP/1.1 200 OK
Content-Length: 44604
Content-Type: application/epub+zip
Last-Modified: Wed, 16 Oct 2019 15:25:34 GMT
ETag: 0x8D7524D16C2A75E
Vary: Origin
Server: nginx
x-ms-request-id: 2674898d-001e-0014-087d-a39e5f000000
x-ms-version: 2009-09-19
x-ms-lease-status: unlocked
x-ms-blob-type: BlockBlob
X-Deity: web03
Date: Mon, 25 Nov 2019 10:47:35 GMT

$ 
```